metadata-ingestion/transformers.md

```diff
@@ -0,0 +1,238 @@
+# MCE-related classes
+from datahub.metadata.schema_classes import (
+    DatasetSnapshotClass,
+    MetadataChangeEventClass,
+    OwnerClass,
+    OwnershipClass,
+    OwnershipTypeClass,
+)
+```
+
+Next, let's define the base scaffolding for the class:
+
+```python
+# append this to the end of custom_transform_example.py
+
+class AddCustomOwnership(Transformer):
+    """Transformer that adds owners to datasets according to a callback function."""
+
+    # context param to generate run metadata such as a run ID
+    ctx: PipelineContext
+    # as defined in the previous block
+    config: AddCustomOwnershipConfig
+
+    def __init__(self, config: AddCustomOwnershipConfig, ctx: PipelineContext):
+        self.ctx = ctx
+        self.config = config
+
+        self.owners = [
+            OwnerClass(owner=owner, type=OwnershipTypeClass.DATAOWNER)
+            for owner in json.loads(config.owner_file)
+        ]
+```
+
+A transformer must have two functions: a `create()` function for initialization and a `transform()` function for executing the transformation.
+
+Let's begin by adding a `create()` method for parsing our configuration dictionary:
+
+```python
+# add this as a function of AddCustomOwnership
+
+@classmethod
+def create(cls, config_dict: dict, ctx: PipelineContext) -> "AddCustomOwnership":
+    config = AddCustomOwnershipConfig.parse_obj(config_dict)
+    return cls(config, ctx)
+```
+
+Now we need to add a `transform()` method that does the work of adding our custom ownership classes. This method will take an MCE as input and output the transformed MCE. Let's offload the processing of each MCE to another `transform_one()` class.
+
+```python
+# add this as a function of AddCustomOwnership
+
+def transform(
+    self, record_envelopes: Iterable[RecordEnvelope]
+) -> Iterable[RecordEnvelope]:
+
+    # loop over envelopes
+    for envelope in record_envelopes:
+
+        # if envelope is an MCE, add the ownership classes
+        if isinstance(envelope.record, MetadataChangeEventClass):
+            envelope.record = self.transform_one(envelope.record)
+        yield envelope
+```
+
+With the main `transform()` method set up, the `transform_one()` method will take a single MCE and add the owners that we loaded from the JSON.
+
+```python
+# add this as a function of AddCustomOwnership
+
+def transform_one(self, mce: MetadataChangeEventClass) -> MetadataChangeEventClass:
+    if not isinstance(mce.proposedSnapshot, DatasetSnapshotClass):
+        return mce
+
+    owners_to_add = self.owners
+
+    if owners_to_add:
+        ownership = builder.get_or_add_aspect(
+            mce,
+            OwnershipClass(
+                owners=[],
+            ),
+        )
+        ownership.owners.extend(owners_to_add)
+
+    return mce
+```
+
+### Installing the package
+
+Now that we've defined the transformer, we need to make it visible to DataHub. This can be done by making sure the Python file is available as a local import.
```

Comment:
add an example codeblock/setup of what this might look like

people skim and copy-paste these tutorials, so we should also emphasize that the setup() method is optional